hls4ml/templates/oneapi/firmware/nnet_utils/nnet_merge_stream.h

```diff
@@ -131,7 +131,7 @@ template <class input1_pipe, class input2_pipe, class res_pipe, typename CONFIG_
         #pragma unroll
         for (int j = 0; j < outputSize; j++) {
             out_data[j] = static_cast<typename ExtractPipeType<res_pipe>::value_type::value_type>(
-                (in_data1[j] < in_data2[j]) ? static_cast<res_T>(in_data1[j]) : static_cast<res_T>(in_data2[j]););
+                (in_data1[j] < in_data2[j]) ? static_cast<res_T>(in_data1[j]) : static_cast<res_T>(in_data2[j]));
```

Comment:
`res_T` in unknown here, so I think the `in_data` needs to be cast directly into the `typename ExtractPipeType<res_pipe>::value_type::value_type` for this to work.